src/lib/routers/history.ts

```diff
@@ -180,6 +202,8 @@ class BrowserHistory<TRouteState> implements Router<TRouteState> {
    * Removes the event listener and cleans up the URL.
    */
   public dispose(): void {
+    this.isSubscribed = false;
```

Comment:
This is nitpicking, so feel free to leave it as is. There are no mentions related to subscription other than here, and I wonder if `this.isDisposed` wouldn't be more appropriate here.

Reply:
I was thinking that isDisposed sounds like a negative condition, but it's actually what we want to know in the later condition, so it can make sense that way too

Reply:
I'll make the update, thanks for the suggestion!